### ISSUE TITLE
some events are decoded as null

### ISSUE THREAD

**Issue by hayeah** [qtumproject/qtumjs#4]
see: https://github.com/qtumproject/qtumjs/issues/2#issuecomment-357575599

**Comment by pplam** [comment]
The case is, I have two contracts, i.e. ContractA and ContractB. When I call methodA of ContractA, methodA calls methodB of ContractB, then all events triggered in methodB are logged to ContractA(doesn't logged to ContractB as ethereum). And all these events are deceded to null.

**Comment by hayeah** [comment]
hmmmm... ok. I don't really have a nice API in mind to handle this case automatically. Maybe you could suggest one.

To fix this problem manually, you can consider merging ContractA and ContractB's ABI arrays:

```
new Contract(rpc, {
   address: "...",
   senderAddress: "...",
   abi: contractA.abi.concat(contractB.abi)
})
```

you could also add only ContractB's events:

```
new Contract(rpc, {
   address: "...",
   senderAddress: "...",
   abi: contractA.abi.concat(contractB.abi.filter(def => def.type === "event"))
})
```

**Comment by hayeah** [comment]
@pplam hi, this issue is resolved in version 1.7.1

A new API is created so that Contract instances are configured with log decoder that can parse all known event types.

For details, see: https://qtumproject.github.io/qtumjs-doc/#contract

This version introduced minor backward incompatible changes with 1.6.*. The only user-visible change, I believe, is that the hexadecimal output are now all consistently without the `0x` prefix.

**Comment by pplam** [comment]
Good job, congratulate!
